### ISSUE TITLE
Enabling subscription to broker store updates breaks Zeek test

### ISSUE THREAD

**Issue by 0xxon** [zeek/broker#135]
After a bit of digging, I noticed that just enabling a subscription to broker store updates (and then discarding the messages) seems to break a test on the Zeek side.

I will still have to dig deeper - but I assume that this is not supposed to happen.

For demonstration - use the ``topic/johanna/brokerstore-subscribing-test`` branch of Zeek. Or cherry-pick the following commit: https://github.com/zeek/zeek/commit/8adc9acec088e0bf2c5712575bdfba26d1b14d36

After that, the known-services btest will fail:
```
$ btest -d scripts.policy.protocols.conn.known-services
[  0%] scripts.policy.protocols.conn.known-services ... failed
  % 'btest-diff knownservices-local.log' failed unexpectedly (exit code 1)
  % cat .diag
  == File ===============================
  #separator \x09
  #set_separator	,
  #empty_field	(empty)
  #unset_field	-
  #path	known_services
  #open	2020-07-13-19-26-58
  #fields	ts	host	port_num	port_proto	service
  #types	time	addr	port	enum	set[string]
  1308930691.089263	172.16.238.131	22	tcp	SSH
  1308930694.550308	172.16.238.131	80	tcp	HTTP
  1594668418.096877	172.16.238.2	53	udp	DNS
  1594668418.096877	172.16.238.131	21	tcp	(empty)
  1594668418.096877	172.16.238.131	80	tcp	(empty)
  #close	2020-07-13-19-26-58
  == Diff ===============================
  --- /tmp/test-diff.28704.knownservices-local.log.baseline.tmp	2020-07-13 19:26:58.000000000 +0000
  +++ /tmp/test-diff.28704.knownservices-local.log.tmp	2020-07-13 19:26:58.000000000 +0000
  @@ -9,5 +9,6 @@
   XXXXXXXXXX.XXXXXX	172.16.238.131	22	tcp	SSH
   XXXXXXXXXX.XXXXXX	172.16.238.131	80	tcp	HTTP
   XXXXXXXXXX.XXXXXX	172.16.238.2	53	udp	DNS
  -XXXXXXXXXX.XXXXXX	172.16.238.131	21	tcp	FTP
  +XXXXXXXXXX.XXXXXX	172.16.238.131	21	tcp	(empty)
  +XXXXXXXXXX.XXXXXX	172.16.238.131	80	tcp	(empty)
   #close XXXX-XX-XX-XX-XX-XX
  =======================================

  % cat .stderr

1 of 1 test failed
```

@Neverlord - if you have any idea what could cause this, that would be awesome.

**Comment by 0xxon** [comment]
This actually breaks much more - the external test suites (e.g. m57-short) have absolutely massive amounts of changes to the baseline. Which is a bit weird - this should not even exercise broker a lot.

However, if you remove the ```bstate->subscriber.add_topic(broker::topics::store_events, true);``` line from the change, everything is back to normal.

**Comment by jsiwek** [comment]
I don't have a complete explanation of the entire cascade of events that ends up causing this strange behavior, but guessing there's nothing really related to the broker-side implementation of store-events, rather just this on the Zeek-side is generally problematic (current-time isn't appropriate/expected for offline analysis):

https://github.com/zeek/zeek/blame/master/src/broker/Manager.cc#L871-L875

Removing that at least restores expected behavior for test cases, so will look a bit more into what the real patch needs to be.

**Comment by Neverlord** [comment]
> will look a bit more into what the real patch needs to be.

@jsiwek let me know if something on the Broker side still comes up. :)

**Comment by jsiwek** [comment]
The bug was in Zeek and fixed in zeek/zeek#1061: https://github.com/zeek/zeek/blob/c3f4971eb25ab47efc827db7977e0d4d05f27bf5/src/broker/Manager.cc#L925-L926
